ui/src/containers/VisualiseResults/SourceResults.js

```diff
@@ -69,21 +76,43 @@ const formatKeyToFriendlyString = (key) => {
   return JSON.stringify(key, null, 2);
 };
 
-const getAxisLabel = (axis, visualisation, type) => {
-  if (type !== XVSY) {
-    return getAxesString(axis, visualisation, type, false);
+const getGroupAxisLabel = (visualisation) => {
+  switch (visualisation.get('type')) {
+    // Correlation Chart type
+    case XVSY:
+      return visualisation.getIn(['axesgroup', 'searchString']) || 'Group';
+    // Bar Chart type
+    case LEADERBOARD:
+    case TEMPLATE_MOST_ACTIVE_PEOPLE:
+    case TEMPLATE_MOST_POPULAR_ACTIVITIES:
+    case TEMPLATE_MOST_POPULAR_VERBS:
+      return visualisation.get('axesyLabel') || visualisation.getIn(['axesgroup', 'searchString']) || 'Y Axis';
+    // Line Chart type
+    case FREQUENCY:
+    case TEMPLATE_ACTIVITY_OVER_TIME:
+      return visualisation.get('axesxLabel') || 'yyyy/mm/dd';
+    default:
+      return visualisation.get('axesxLabel') || visualisation.getIn(['axesgroup', 'searchString']) || 'X Axis';
   }
-  return visualisation.getIn(['axesgroup', 'searchString'], 'No value');
 };
 
-const createSelectIfXVSY = (index, visualisation, type, axis) => {
-  if (type !== XVSY) {
-    return getAxisLabel(axis, visualisation, type);
+const getValueAxisLabel = (index, visualisation) => {
```

Comment:
I renamed this function. Source Results show group axis in most left column and show value axis in right columns.